pycloudlib/lxd/instance.py

```diff
@@ -379,6 +379,32 @@ def wait_for_stop(self):
             time.sleep(1)
         raise TimeoutError
 
+    def _wait_for_instance_start(self):
```

Comment:
That's unfortunate. I am also on 4.14 (latest/stable)

Curious what happens when you try `lxc list -c N <vm_name> -f csv` while the test is waiting for the VM. Does it ever return anything meaningful? Are you able to shell/ssh to it while the command isn't returning valid info?

I'm assuming you're using the `pycloudlib-vm-bionic-v2` profile? It works for me when I use that profile.

Reply:
@lucasmoura @TheRealFalcon I played around with uaclient tests and this these pycloudlib changes a bit and discovered a couple things.

1. `tox -e behave-vm-20.04` succeeds for uaclient with these changes
2. `tox -e behave-vm-18.04` times out with these changes due to this code when bringing up the temporary image-build vm
3. `tox -e behave-vm-18.04` _without_ these changes works but the temporary image-build vm never gets a positive process count.
    a. I was unable to `lxc exec` anything on the instance manually, but, weirdly, our test code is able to successfully run commands on the instance. 
    b. later in the same test, it brings up a second vm. The second vm does get a positive process count and I can `lxc exec` on it.
    c. the difference between our commands to bring up the image-build vm and the second vm are that we include some cloud-init user-data for the image-build vm. That must be breaking something

So I think this might count as a bug in our uaclient tests :upside_down_face: and these changes are probably fine.

Reply:
I confirmed that the test failures should be considered a bug in uaclient - not this PR - and I put up a uaclient PR to fix it: https://github.com/canonical/ubuntu-advantage-client/pull/1665